apps/cowswap-frontend/src/modules/volumeFee/state/safeAppFeeAtom.ts

```diff
@@ -0,0 +1,66 @@
+import { atom } from 'jotai'
+
+import { STABLECOINS } from '@cowprotocol/common-const'
+import { getCurrencyAddress } from '@cowprotocol/common-utils'
+import { OrderKind } from '@cowprotocol/cow-sdk'
+import { walletDetailsAtom, walletInfoAtom } from '@cowprotocol/wallet'
+
+import { derivedTradeStateAtom } from '../../trade'
+import { VolumeFee } from '../types'
+
+const SAFE_FEE_RECIPIENT = '0x63695Eee2c3141BDE314C5a6f89B98E62808d716'
+
+const FEE_TIERS = {
+  TIER_1: 100_000, // 0 - 100k
+  TIER_2: 1_000_000, // 100k - 1m
+}
+
+const FEE_PERCENTAGE_BPS = {
+  REGULAR: {
+    TIER_1: 35,
+    TIER_2: 20,
+    TIER_3: 10,
+  },
+  STABLE: {
+    TIER_1: 10,
+    TIER_2: 7,
+    TIER_3: 5,
+  },
+}
+
+/**
+ * https://help.safe.global/en/articles/178530-how-does-the-widget-fee-work-for-native-swaps
+ * https://github.com/safe-global/safe-wallet-web/blob/0818e713fa0f9bb7a6472e34a05888896ffc3835/src/features/swap/helpers/fee.ts
+ */
+export const safeAppFeeAtom = atom<VolumeFee | null>((get) => {
+  const { chainId } = get(walletInfoAtom)
+  const { isSafeApp } = get(walletDetailsAtom)
```

Comment:
will this affect also wallet connect, or just the safe app?

Reply:
Just Safe App